### PR TITLE
Creating SenservaPro Data Connector, Workbooks, and Queries

### PR DESCRIPTION
**Pull request by tj-senserva**
## Proposed Changes

  - Creating SenservaPro Data Connector, Workbooks, and Queries

### CODE REVIEW

**Comment by preetikr** [comment]
@tj-senserva - Thanks, for your contribution! Would you be interested in delivering this as Solutions? Packaged content easily discoverable and deployable?

**Comment by tj-senserva** [comment]
@preetikr - Thank you. Yes, we would be interested, we had built this in mind to be part of a Solution.

**Comment by preetikr** [comment]
@tj-senserva - Cool - then can you please update the PR to make a folder under https://github.com/Azure/Azure-Sentinel/tree/master/Solutions for SenservaPro and create sub folders like Data Connector, Parser, Hunting Queries, etc.? Refer to https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/McAfeeePO as an example. Thanks.

File: DataConnectors/Senserva/SenservaPro.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
+{
+	"id": "SenservaPro",
+	"title": "SenservaPro",
+	"publisher": "Senserva",
+	"descriptionMarkdown": "The SenservaPro data connector provides a viewing experience for your SenservaPro scanning logs. View dashboards of your data, use queries to hunt & explore, and create custom alerts for your needs. Start your security journey today",
+	"additionalRequirementBanner": "No additional requirements needed",
+	"graphQueries": [
+		{
+			"metricName": "Total data received",
+			"legend": "SenservaPro",
```
Comment (v-admahe) [inline DataConnectors/Senserva/SenservaPro.json:10]: Legend name should be data type name. Please update the legend to SenservaPro_CL.

Thanks,
Aditi Maheshwari

File: DataConnectors/Senserva/SenservaPro.json [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	"connectivityCriterias": [
+		{
+			"type": "IsConnectedQuery",
+			"value": [
+				"SenservaPro_CL\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+			]
+		}
+	],
+	"availability": {
+		"status": 0,
```
Comment (v-admahe) [inline DataConnectors/Senserva/SenservaPro.json:103]: Status should be 2 .Please update the status.

Thanks,
Aditi Maheshwari

File: DataConnectors/Senserva/SenservaPro.json [hunk 3]
```diff
@@ -0,0 +1,159 @@
+{
+	"id": "SenservaPro",
+	"title": "SenservaPro",
+	"publisher": "Senserva",
+	"descriptionMarkdown": "The SenservaPro data connector provides a viewing experience for your SenservaPro scanning logs. View dashboards of your data, use queries to hunt & explore, and create custom alerts for your needs. Start your security journey today",
+	"additionalRequirementBanner": "No additional requirements needed",
```
Comment (v-admahe) [inline DataConnectors/Senserva/SenservaPro.json:6]: If there is no additional requirement. please remove this tag.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-admahe** [comment]
@tj-senserva : Workbook (SenservaProAnalyticsWorkbook.json)queries are not appropriate. Could you please make the required change.

image
image
image

Thanks,
Aditi Maheshwari

**Comment by v-admahe** [comment]
@tj-senserva Could you please resolve the merge conflicts.

Thanks,
Aditi Maheshwari

**Comment by tj-senserva** [comment]
@preetikr - Thank you, the Solutions Folder has been created now

File: DataConnectors/Senserva/SenservaPro.json (hunk 1 above)
Comment (tj-senserva) [reply DataConnectors/Senserva/SenservaPro.json]: @v-admahe This change has been made

File: DataConnectors/Senserva/SenservaPro.json (hunk 2 above)
Comment (tj-senserva) [reply DataConnectors/Senserva/SenservaPro.json]: @v-admahe This change has been made

File: DataConnectors/Senserva/SenservaPro.json (hunk 3 above)
Comment (tj-senserva) [reply DataConnectors/Senserva/SenservaPro.json]: @v-admahe This change has been made

**Comment by tj-senserva** [comment]
@v-admahe The merge conflicts have been resolved

**Comment by tj-senserva** [comment]
@v-admahe 
This query returned data for me, but I've gone ahead and changed the time range to constant values


image

File: Workbooks/SenservaProMultipleWorkspaceWorkbook.json [hunk 4]
```diff
@@ -0,0 +1,132 @@
+{
+  "version": "Notebook/1.0",
+  "items": [
+    {
+      "type": 1,
+      "content": {
+        "json": "## Using SenservaPro Across Multiple Log Analytics Workspaces\n---\n\nThis workbook helps serve as a guide for users who are using SenservaPro across multiple instances of SenservaPro. These users generally have multiple tenants or a geographical data location requirement. \nMore information can be found at https://docs.microsoft.com/en-us/azure/sentinel/extend-sentinel-across-workspaces-tenants\n\n\nSelect one workspace as the main conduit to create a unified view. Create a Sentinel Function to provide an easy alias for your queries.\nMore information about Sentinel Functions can be found at https://docs.microsoft.com/en-us/azure/azure-monitor/logs/functions"
```
Comment (v-admahe) [inline Workbooks/SenservaProMultipleWorkspaceWorkbook.json:7]: In the link country code (en-us) should not mention due to this checks are failing. Please make the required change.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-admahe** [comment]
> @tj-senserva - Cool - then can you please update the PR to make a folder under https://github.com/Azure/Azure-Sentinel/tree/master/Solutions for SenservaPro and create sub folders like Data Connector, Parser, Hunting Queries, etc.? Refer to https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/McAfeeePO as an example. Thanks.

@tj-senserva : you don't have to copy paste those file, you have to move those file under solution folder . Please move the files into the solution folder. Please refer to https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/McAfeeePO as an example. 

Thanks,
Aditi Maheshwari

**Comment by v-admahe** [comment]
> @tj-senserva : Workbook (SenservaProAnalyticsWorkbook.json)queries are not appropriate. Could you please make the required change.

@tj-senserva : In workbook (SenservaProMultipleWorkspaceWorkbook) queries the coulmn name used as "Tenant_ExportWorkspaceDisplayName_s" , coulmn is not available in the sample data available in the PR. Could you please add this coulmn in sample data.

image

 In workbook (SenservaProMultipleWorkspaceWorkbook) queries the coulmn name used as Severity but it should be Severity_s.Could you please update the query.

image

Thanks,
Aditi Maheshwari

**Comment by tj-senserva** [comment]
> > @tj-senserva - Cool - then can you please update the PR to make a folder under https://github.com/Azure/Azure-Sentinel/tree/master/Solutions for SenservaPro and create sub folders like Data Connector, Parser, Hunting Queries, etc.? Refer to https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/McAfeeePO as an example. Thanks.
> 
> @tj-senserva : you don't have to copy paste those file, you have to move those file under solution folder . Please move the files into the solution folder. Please refer to https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/McAfeeePO as an example.
> 
> Thanks,
> Aditi Maheshwari

@v-admahe Can you please clarify? The McAfeeePO artifacts exists only in the Solution folder, but CiscoUmbrella artifacts exist in both Solution folder (https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/CiscoUmbrella/Workbooks/CiscoUmbrella.json) and outside (https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks/CiscoUmbrella.json). 

Can you please confirm that I remove my workbooks entry from the WorkbooksMetadata.json? I do not see entries for McAfeeePO there. (https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks/WorkbooksMetadata.json)

File: Workbooks/SenservaProMultipleWorkspaceWorkbook.json (hunk 4 above)
Comment (tj-senserva) [reply Workbooks/SenservaProMultipleWorkspaceWorkbook.json]: @v-admahe - This change has been made

**Comment by tj-senserva** [comment]
> > @tj-senserva : Workbook (SenservaProAnalyticsWorkbook.json)queries are not appropriate. Could you please make the required change.
> 
> @tj-senserva : In workbook (SenservaProMultipleWorkspaceWorkbook) queries the coulmn name used as "Tenant_ExportWorkspaceDisplayName_s" , coulmn is not available in the sample data available in the PR. Could you please add this coulmn in sample data.
> 
> In workbook (SenservaProMultipleWorkspaceWorkbook) queries the coulmn name used as Severity but it should be Severity_s.Could you please update the query.
> 
> Please check this field in other workbook as well.
> 
> Thanks,
> Aditi Maheshwari

@v-admahe - The sample data has been updated with the field. All queries should be using 'Severity' now

**Review by shainw — CHANGES_REQUESTED** [review]
Query Changes:

**General for all Detections/Analytic Rules**
Please remove Remove ```| where TimeGenerated...```, the detection will follow the queryPeriod which is set to 6h on each. Generally, you only need to include timeframes when the query is going to utilize multiple lookback periods for baselines and such.  Also, for any detection, the max default lookback time is 14d for performance reasons.

**General for all Hunting Queries**
60d is longer than recommend for a default for Hunting queries due to potential performance issues, max is 30d as a default.

File: Detections/SenservaPro/AccountDisabled.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+id: 24E0132F-61D1-41BD-9393-06136D1039C7
+name: Disabled
```
[the remark this answers is not in the document]
Comment (tj-senserva) [reply Detections/SenservaPro/AccountDisabled.yaml]: @shainw - This name has been updated to be more descriptive

**Comment by tj-senserva** [comment]
> Query Changes:
> 
> **General for all Detections/Analytic Rules**
> Please remove Remove `| where TimeGenerated...`, the detection will follow the queryPeriod which is set to 6h on each. Generally, you only need to include timeframes when the query is going to utilize multiple lookback periods for baselines and such. Also, for any detection, the max default lookback time is 14d for performance reasons.

@shainw - The Detection queries have had the `| where TimeGenerated...` field removed.

> **General for all Hunting Queries**
> 60d is longer than recommend for a default for Hunting queries due to potential performance issues, max is 30d as a default.

The Hunting Queries have had the TimeGenerated reduced from 60d to 14d

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: Detections/SenservaPro/AccountDisabled.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+description: |
+  'This query searches for account is Disabled. Does not effect score as its easily enabled.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: SenservaPro
+    dataTypes:
+      - SenservaPro
```
[the remark this answers is not in the document]
Comment (tj-senserva) [reply Detections/SenservaPro/AccountDisabled.yaml]: @v-jayakal - This change has been made

**Review by shainw — COMMENTED** [review]
Queries are approved

**Comment by v-maudan** [comment]
@tj-senserva , since 2 sets of files are not needed, we removed workbookmmetadata.json changes as well as other location files and  kept all changes in solution folder. Thanks!.

**Comment by tj-senserva** [comment]
> @tj-senserva , since 2 sets of files are not needed, we removed workbookmetadata.json changes as well as other location files and kept all changes in solution folder. Thanks!.

@v-maudan - Thanks for yours and everyone's help with this

**Pull request merged**